providers/BaseProviders/src/main/java/net/ess3/provider/providers/ModernPersistentDataProvider.java

```diff
@@ -0,0 +1,43 @@
+package net.ess3.provider.providers;
+
+import net.ess3.provider.PersistentDataProvider;
+import org.bukkit.NamespacedKey;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.meta.ItemMeta;
+import org.bukkit.persistence.PersistentDataType;
+import org.bukkit.plugin.Plugin;
+
+@SuppressWarnings("ConstantConditions")
+public class ModernPersistentDataProvider implements PersistentDataProvider {
+    private final Plugin plugin;
+
+    public ModernPersistentDataProvider(Plugin plugin) {
+        this.plugin = plugin;
+    }
+
+    @Override
+    public void set(ItemStack itemStack, String key, String value) {
+        final ItemMeta im = itemStack.getItemMeta();
+        im.getPersistentDataContainer().set(new NamespacedKey(plugin, key), PersistentDataType.STRING, value);
+        itemStack.setItemMeta(im);
+    }
+
+    @Override
+    public String getString(ItemStack itemStack, String key) {
+        try {
+            return itemStack.getItemMeta().getPersistentDataContainer().get(new NamespacedKey(plugin, key), PersistentDataType.STRING);
+        } catch (NullPointerException | IllegalArgumentException ignored) {
```

Comment:
Would it not be better to check `itemStack.getMaterial().isAir()` instead of catching NPE here?

Reply:
Yes but no!